### PR TITLE
Gateway API doc improvements

### PR DESCRIPTION
**Pull request by frankbu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md [hunk 1]
```diff
@@ -14,17 +14,19 @@ This task describes how to configure Istio to expose a service outside the servi
...
-{{< /warning >}}
-
 ## Setup
 
 1. The Gateway APIs do not come installed by default on most Kubernetes clusters. Install the Gateway API CRDs if they are not present:
 
     {{< text bash >}}
-    $ kubectl get crd gateways.gateway.networking.k8s.io || { kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.0" | kubectl apply -f -; }
+    $ kubectl get crd gateways.gateway.networking.k8s.io || \
+      { kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.5.0" | kubectl apply -f -; }
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md:23]: Note that I've bumped the version from v0.4.0 to v0.5.0

File: content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md [hunk 2]
```diff
@@ -14,17 +14,19 @@ This task describes how to configure Istio to expose a service outside the servi
 These APIs are an actively developed evolution of the Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/)
 and [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) APIs.
 
-{{< warning >}}
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md:17]: Do we still need this, or some different, warning now?

Comment (howardjohn) [reply content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md:17]: Can we put it only in the mesh section?

Comment (howardjohn) [reply content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md:17]: Maybe linking to https://gateway-api.sigs.k8s.io/contributing/gamma/ and say its pending upstream agreement

**Review by howardjohn — COMMENTED** [review]
Thanks!

**Review by howardjohn — APPROVED** [review]

File: content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md (hunk 2 above)
Comment (frankbu) [reply content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md:17]: Sounds good. I also added a couple more steps to test add header.

**Review by craigbox — APPROVED** [review]
Thanks both!

**Review by ericvn — APPROVED** [review]

**Pull request merged**